src/Model/AutoSegmentation.py

```diff
@@ -0,0 +1,180 @@
+import os
+import shutil
+import tempfile
+import logging
+from src.Model.PatientDictContainer import PatientDictContainer
+from src.Model.Worker import SegmentationWorkerSignals
+import fnmatch
+from totalsegmentator.python_api import totalsegmentator
+from src.Model.NiftiToRtstructConverter import nifti_to_rtstruct_conversion
+from src.View.util.RedirectStdOut import ConsoleOutputStream, redirect_output_to_gui, setup_logging
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+logger = logging.getLogger(__name__)
+
+
+class AutoSegmentation:
+    """Handles the automatic segmentation process.
+
+    This class manages the workflow for running TotalSegmentator, including
+    copying DICOM files, setting up the segmentation task, and converting the
+    output to DICOM RTSTRUCT format.
+    """
+
+    def __init__(self, controller):
+        self.controller = controller
+        self.patient_dict_container = PatientDictContainer()
+        self.dicom_dir = self.patient_dict_container.path  # Get the current loaded dir to DICOM series
+        self.dicom_temp_dir = None
+
+        self.signals = SegmentationWorkerSignals()
+
+        # Connect worker signals to controller slots
+        self.signals.progress_updated.connect(self.controller.update_progress_text)
+        self.signals.finished.connect(self.controller.on_segmentation_finished)
+        self.signals.error.connect(self.controller.on_segmentation_error)
+
+    def _connect_terminal_stream_to_gui(self) -> None:
```

Comment:
This looks like it was done very similarly elsewhere.  I'm not saying it must be consolidated, but if it makes sense to do so, please consider it.